src/App.js

```diff
@@ -1,8 +1,36 @@
 import React from 'react';
 import './App.css';
+// import ChatEntry from './components/ChatEntry';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+
+  const likeEntry = (id) => {
+    setEntries((entries) => {
+      return entries.map((entry) => {
+        if (entry.id === id) {
+          return {
+            ...entry,
+            liked: !entry.liked,
+          };
+        } else {
+          return entry;
+        }
+      });
+    });
+  };
```

Comment:
Great idea to map each entry in the array. I love the way you did all of this! If you wanted to cut down a bit on the nesting, you could have set a variable called newEntries to the return and then used that variable within the setEntries function but overall this works well!